uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -378,8 +378,10 @@ define([
      */
     function leaveRequestDays (leaveRequest) {
       var days = [];
-      var pointerDate = moment(leaveRequest.from_date).clone();
-      var toDate = moment(leaveRequest.to_date);
+      var pointerDate = moment(leaveRequest.from_date)
+        .hour(0).minute(0).second(0).clone();
```

Comment:
```js
var pointerDate = moment(leaveRequest.from_date).startOf('day');
var toDate = moment(leaveRequest.to_date).startOf('day');
```

Side question: Why do you need to clone it if you already modified it?

Reply:
No idea, I also think the .clone() is not needed here.

Reply:
It actually goes from here https://github.com/civicrm/civihr/pull/1989, but I also don't think `clone` is needed here. We instantiate a `moment` object from a string and then use `format()` as output. So we don't store references.